examples/ProjPlaneCurve.jl

```diff
@@ -0,0 +1,534 @@
+module ProjPlaneCurveModule
+
+using Oscar, Markdown
+
+include("AffinePlaneCurve.jl")
+using .AffinePlaneCurveModule
+
+import Base.:(==)
+
+export ProjPlaneCurve, degree, jacobi_ideal, issmooth, tangent,
+       curve_components, isirreducible, isreduced, common_component, reduction,
+       dehomogenization, curve_intersect, curve_singular_locus, issmooth_curve
+
+################################################################################
+#
+# We follow the lecture notes by Andreas Gathmann:
+# https://www.mathematik.uni-kl.de/~gathmann/en/curves.php
+#
+################################################################################
+#
+
+mutable struct ProjPlaneCurve{S}
+  eq::Oscar.MPolyElem_dec{S}              # Equation of the curve (polynomial in three variables)
+  degree::Int64                           # degree of the equation of the curve
+  dimension::Int64                        # dimension of the curve (as a variety)
+  components::Dict
+  function ProjPlaneCurve(eq::Oscar.MPolyElem_dec{S}) where {S <: FieldElem}
+    r = new{S}()
+    r.eq = eq
+    if nvars(parent(r.eq)) != 3
+       error("The defining equation must belong to a ring with three variables")
+    elseif isconstant(r.eq) == true
+       error("The defining equation must be non constant")
+    else
+       if ishomogenous(r.eq) == false
+          error("The defining equation is not homogeneous")
+       end
+    end
+    r.degree = -1                      # -1 when it is not computed yet
+    r.dimension = 1                    # since C is a plane curve, the dimension is always 1
+    r.components = Dict{ProjPlaneCurve{S}, Int64}()
+    return r
```

Comment:
```suggestion
    if nvars(parent(eq)) != 3
       error("The defining equation must belong to a ring with three variables")
    elseif isconstant(r.eq)
       error("The defining equation must be non constant")
    elseif !ishomogenous(eq)
       error("The defining equation is not homogeneous")
    end
    new{S}(eq, 
           -1, # -1 when it is not computed yet
           1,  # since C is a plane curve, the dimension is always 1
           Dict{ProjPlaneCurve{S}, Int64}())
```